lib/lib.es5.d.ts

```diff
@@ -236,7 +236,7 @@ interface ObjectConstructor {
       * Returns the names of the enumerable properties and methods of an object.
       * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
       */
-    keys(o: any): string[];
+    keys<A>(o: A): (keyof A)[];
```

Comment:
`<A extends {}>`

The problem with making `Object.keys` return `keyof A` is that typescript makes no guarantees that the input object has _only_ the keys that are specified in its type. There may be valid values in the array that are not considered by the type.